src/config/plugin.ts

```diff
@@ -284,25 +335,77 @@ export class Plugin implements IPlugin {
   }
 
   private async getCommandIDs(): Promise<string[]> {
-    const commandsDir = await this.getCommandsDir()
-    if (!commandsDir) return []
-
     const marker = Performance.mark(OCLIF_MARKER_OWNER, `plugin.getCommandIDs#${this.name}`, {plugin: this.name})
-    this._debug(`loading IDs from ${commandsDir}`)
-    const files = await globby(GLOB_PATTERNS, {cwd: commandsDir})
-    const ids = processCommandIds(files)
+
+    let ids: string[]
+    switch (this.commandDiscoveryOpts?.strategy) {
+      case 'explicit': {
+        ids = (await this.getCommandIdsFromTarget()) ?? []
+        break
+      }
+
+      case 'pattern': {
+        ids = await this.getCommandIdsFromPattern()
+        break
+      }
+
+      case 'single': {
+        ids = (await this.getCommandIdsFromTarget()) ?? []
+        break
+      }
+
+      default: {
+        ids = []
+      }
+    }
+
     this._debug('found commands', ids)
     marker?.addDetails({count: ids.length})
     marker?.stop()
     return ids
   }
 
+  private async getCommandIdsFromPattern(): Promise<string[]> {
+    const commandsDir = await this.getCommandsDir()
+    if (!commandsDir) return []
+
+    this._debug(`loading IDs from ${commandsDir}`)
+    const files = await globby(this.commandDiscoveryOpts?.globPatterns ?? GLOB_PATTERNS, {cwd: commandsDir})
+    return processCommandIds(files)
+  }
+
+  private async getCommandIdsFromTarget(): Promise<string[] | undefined> {
+    const commandsFromExport = await this.loadCommandsFromTarget()
+    if (commandsFromExport) {
+      return Object.keys(commandsFromExport)
+    }
+  }
+
   private async getCommandsDir(): Promise<string | undefined> {
     if (this.commandsDir) return this.commandsDir
-    this.commandsDir = await tsPath(this.root, this.pjson.oclif.commands, this)
+
+    this.commandsDir = await tsPath(this.root, this.commandDiscoveryOpts?.target, this)
     return this.commandsDir
   }
 
+  private async loadCommandsFromTarget(): Promise<Record<string, Command.Class> | undefined> {
```

Comment:
you defined type `CommandCache` above, so   

```suggestion
  private async loadCommandsFromTarget(): Promise<CommandCache | undefined> {
```